stotko/stdgpu
# CI: Simplify job structure

Our CI job structure contains a lot of repetitions and is requires quite a lot of maintenance to keep it up to date. Simplify the structure:

- Unify workflows:
    - `Ubuntu` + `Windows` -> `Tests` 
    - `Analysis` + `Style` -> `Lint`
- Add dependabot to periodically check for updates
- Add jobs to check if all jobs in a workflow succeeded
- Allow canceling jobs to speed up the re-runs
